rqt_bag/src/rqt_bag/recorder.py

```diff
@@ -223,39 +294,54 @@ def _record(self, topic, m):
             self._unsubscribe(topic)
             return
 
-        self._write_queue.put((topic, m, rospy.get_rostime()))
+        now = Clock(clock_type=ClockType.SYSTEM_TIME).now()
+        self._write_queue.put((topic, msg, msg_type_name, now))
         self._message_count[topic] += 1
 
     def _run_write(self):
         try:
+            poll_interval = 1.0
             while not self._stop_flag:
-                # Wait for a message
-                item = self._write_queue.get()
+                try:
+                    item = self._write_queue.get(block=False)
+                except Empty:
+                    time.sleep(poll_interval)
+                    continue
```

Comment:
Why did this logic need to change? I'm guessing it was a bug related failing to check the stop flag. Should this also be fixed for ROS 1?

Using the queue timer seems like it would be better than sleeping for a fixed period. Then we're writing as soon as data is available.
```suggestion
                try:
                    item = self._write_queue.get(block=True, timeout=poll_interval)
                except Empty:
                    continue
```

Reply:
I think this was introduced by the previous start to the ROS2 port. Updated.